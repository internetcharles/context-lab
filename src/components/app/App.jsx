import React from 'react';
import {
  BrowserRouter as Router, 
  Route, 
  Switch,
} from 'react-router-dom';
import CharacterPage from './pages/CharacterPage';

export default function App() {
  return (
    <div>
      <Router>
        <Switch>
          <Route 
            path="/" 
            exact
            component={CharacterPage} />
        </Switch>
      </Router>
    </div>
  );
}
