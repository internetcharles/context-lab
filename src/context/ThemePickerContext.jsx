import React, { useState } from 'react';

export const ThemePickerContext = React.createContext();

export const ThemePickerContext = ({ children }) => {
  
}