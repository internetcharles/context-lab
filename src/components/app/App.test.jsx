import React from 'react';
import { render, cleanup } from '@testing-library/react';
import App from './App';
import { ThemePickerProvider } from '../../context/ThemePickerContext';

describe('App component', () => {
  afterEach(() => cleanup());
  it('renders App', () => {
    const { asFragment } = render(
      <ThemePickerProvider>
        <App />
      </ThemePickerProvider>);
    expect(asFragment()).toMatchSnapshot();
  });
});
