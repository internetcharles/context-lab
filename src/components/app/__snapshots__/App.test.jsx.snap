// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<DocumentFragment>
  <div
    style="background-color: rgb(0, 0, 0); color: rgb(255, 255, 255);"
  >
    <h1>
      HEY ARNOLD
    </h1>
    <button>
      Change Theme
    </button>
    <button
      disabled=""
    >
      Prev Page
    </button>
    <button>
      Next Page
    </button>
    <h1>
      Loading...
    </h1>
  </div>
</DocumentFragment>
`;
